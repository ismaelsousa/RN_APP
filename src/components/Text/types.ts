/* eslint-disable @typescript-eslint/no-unused-vars */
import React from 'react';
import {TextProps} from 'react-native';
import {ColorsType, TypographyType} from 'styled-components/native';
import {DefaultTheme} from 'styled-components/native';

export interface Props extends TextProps {
  children: string | React.ReactNode;
  color?: ColorsType;
  typography?: TypographyType;
  theme?: DefaultTheme;
}
