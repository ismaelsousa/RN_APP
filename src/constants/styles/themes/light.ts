/* eslint-disable @typescript-eslint/no-unused-vars */
import {DefaultTheme} from 'styled-components/native';
import * as Colors from '../colors';
import {typographyCommon, borders, spacing} from './common';

export const themeLight: DefaultTheme = {
  colors: {
    primary: {
      main: Colors.brand01,
      onMain: Colors.white,
    },
    secondary: {
      main: Colors.brand02,
      onMain: Colors.white,
    },
    error: {
      main: Colors.red,
      onMain: Colors.white,
    },
    background: {
      main: Colors.white,
      onMain: Colors.black,
    },
    surface: {
      main: Colors.black,
      onMain: Colors.white,
    },
    surface50: {
      main: Colors.grey50,
      onMain: Colors.white,
    },
    surface100: {
      main: Colors.grey100,
      onMain: Colors.white,
    },
    surface500: {
      main: Colors.grey500,
      onMain: Colors.white,
    },
    surface600: {
      main: Colors.grey600,
      onMain: Colors.white,
    },
    orange: {
      main: Colors.orange,
      onMain: Colors.white,
    },
    lightGreen: {
      main: Colors.lightGreen,
      onMain: Colors.white,
    },
  },
  typography: typographyCommon,
  borders,
  spacing,
};
