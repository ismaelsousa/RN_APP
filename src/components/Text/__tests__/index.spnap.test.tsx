import React from 'react';
import {render} from '@testing-library/react-native';
import Text from '../';
import {themeLight} from '~/constants/styles/themes/light';
import 'jest-styled-components';

describe('Text Snapshot', () => {
  it('should render without props', () => {
    const tree = render(
      <Text theme={themeLight}>Hello world test</Text>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render with primary color', () => {
    const tree = render(
      <Text theme={themeLight} color="primary">
        Hello world test
      </Text>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render with typography caption', () => {
    const tree = render(
      <Text theme={themeLight} typography="caption">
        Hello world test
      </Text>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
