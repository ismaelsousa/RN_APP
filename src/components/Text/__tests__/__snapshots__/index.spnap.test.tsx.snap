// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Snapshot should render with primary color 1`] = `
<Text
  color="primary"
  style={
    Array [
      Object {
        "color": "#3C9BA6",
        "fontFamily": "OpenSans-Regular",
        "fontSize": 14,
      },
    ]
  }
  theme={
    Object {
      "borders": Object {
        "radius": Object {
          "sm": 10,
          "xs": 4,
        },
      },
      "colors": Object {
        "background": Object {
          "main": "#fff",
          "onMain": "#000",
        },
        "error": Object {
          "main": "#D93936",
          "onMain": "#fff",
        },
        "lightGreen": Object {
          "main": "#95BFA7",
          "onMain": "#fff",
        },
        "orange": Object {
          "main": "#F28705",
          "onMain": "#fff",
        },
        "primary": Object {
          "main": "#3C9BA6",
          "onMain": "#fff",
        },
        "secondary": Object {
          "main": "#072B59",
          "onMain": "#fff",
        },
        "surface": Object {
          "main": "#000",
          "onMain": "#fff",
        },
        "surface100": Object {
          "main": "#4E4B59",
          "onMain": "#fff",
        },
        "surface50": Object {
          "main": "#E1E0E7",
          "onMain": "#fff",
        },
        "surface500": Object {
          "main": "#6B7785",
          "onMain": "#fff",
        },
        "surface600": Object {
          "main": "#41474D",
          "onMain": "#fff",
        },
      },
      "spacing": Object {
        "lg": 25,
        "md": 20,
        "sm": 10,
        "ty": 4,
        "xl": 38,
        "xs": 5,
        "xxl": 45,
        "xxxl": 100,
      },
      "typography": Object {
        "body1": Object {
          "fontFamily": "OpenSans-Regular",
          "fontSize": 14,
        },
        "body2": Object {
          "fontFamily": "OpenSans-Bold",
          "fontSize": 16,
        },
        "body3": Object {
          "fontFamily": "OpenSans-Regular",
          "fontSize": 14,
        },
        "caption": Object {
          "fontFamily": "OpenSans-Regular",
          "fontSize": 16,
        },
        "h1": Object {
          "fontFamily": "OpenSans-Light",
          "fontSize": 96,
        },
        "h2": Object {
          "fontFamily": "Montserrat-Bold",
          "fontSize": 48,
        },
        "h3": Object {
          "fontFamily": "Montserrat-Bold",
          "fontSize": 32,
        },
        "h4": Object {
          "fontFamily": "Montserrat-SemiBold",
          "fontSize": 32,
        },
        "h5": Object {
          "fontFamily": "Montserrat-Bold",
          "fontSize": 24,
        },
        "h6": Object {
          "fontFamily": "Montserrat-Regular",
          "fontSize": 24,
        },
        "h7": Object {
          "fontFamily": "Montserrat-SemiBold",
          "fontSize": 24,
        },
        "h8": Object {
          "fontFamily": "Montserrat-SemiBold",
          "fontSize": 20,
        },
        "overline": Object {
          "fontFamily": "OpenSans-SemiBold",
          "fontSize": 14,
        },
        "subtitle1": Object {
          "fontFamily": "OpenSans-Regular",
          "fontSize": 18,
        },
        "subtitle2": Object {
          "fontFamily": "Montserrat-Bold",
          "fontSize": 18,
        },
      },
    }
  }
  typography="body1"
>
  Hello world test
</Text>
`;

exports[`Text Snapshot should render with typography caption 1`] = `
<Text
  color="surface"
  style={
    Array [
      Object {
        "color": "#000",
        "fontFamily": "OpenSans-Regular",
        "fontSize": 16,
      },
    ]
  }
  theme={
    Object {
      "borders": Object {
        "radius": Object {
          "sm": 10,
          "xs": 4,
        },
      },
      "colors": Object {
        "background": Object {
          "main": "#fff",
          "onMain": "#000",
        },
        "error": Object {
          "main": "#D93936",
          "onMain": "#fff",
        },
        "lightGreen": Object {
          "main": "#95BFA7",
          "onMain": "#fff",
        },
        "orange": Object {
          "main": "#F28705",
          "onMain": "#fff",
        },
        "primary": Object {
          "main": "#3C9BA6",
          "onMain": "#fff",
        },
        "secondary": Object {
          "main": "#072B59",
          "onMain": "#fff",
        },
        "surface": Object {
          "main": "#000",
          "onMain": "#fff",
        },
        "surface100": Object {
          "main": "#4E4B59",
          "onMain": "#fff",
        },
        "surface50": Object {
          "main": "#E1E0E7",
          "onMain": "#fff",
        },
        "surface500": Object {
          "main": "#6B7785",
          "onMain": "#fff",
        },
        "surface600": Object {
          "main": "#41474D",
          "onMain": "#fff",
        },
      },
      "spacing": Object {
        "lg": 25,
        "md": 20,
        "sm": 10,
        "ty": 4,
        "xl": 38,
        "xs": 5,
        "xxl": 45,
        "xxxl": 100,
      },
      "typography": Object {
        "body1": Object {
          "fontFamily": "OpenSans-Regular",
          "fontSize": 14,
        },
        "body2": Object {
          "fontFamily": "OpenSans-Bold",
          "fontSize": 16,
        },
        "body3": Object {
          "fontFamily": "OpenSans-Regular",
          "fontSize": 14,
        },
        "caption": Object {
          "fontFamily": "OpenSans-Regular",
          "fontSize": 16,
        },
        "h1": Object {
          "fontFamily": "OpenSans-Light",
          "fontSize": 96,
        },
        "h2": Object {
          "fontFamily": "Montserrat-Bold",
          "fontSize": 48,
        },
        "h3": Object {
          "fontFamily": "Montserrat-Bold",
          "fontSize": 32,
        },
        "h4": Object {
          "fontFamily": "Montserrat-SemiBold",
          "fontSize": 32,
        },
        "h5": Object {
          "fontFamily": "Montserrat-Bold",
          "fontSize": 24,
        },
        "h6": Object {
          "fontFamily": "Montserrat-Regular",
          "fontSize": 24,
        },
        "h7": Object {
          "fontFamily": "Montserrat-SemiBold",
          "fontSize": 24,
        },
        "h8": Object {
          "fontFamily": "Montserrat-SemiBold",
          "fontSize": 20,
        },
        "overline": Object {
          "fontFamily": "OpenSans-SemiBold",
          "fontSize": 14,
        },
        "subtitle1": Object {
          "fontFamily": "OpenSans-Regular",
          "fontSize": 18,
        },
        "subtitle2": Object {
          "fontFamily": "Montserrat-Bold",
          "fontSize": 18,
        },
      },
    }
  }
  typography="caption"
>
  Hello world test
</Text>
`;

exports[`Text Snapshot should render without props 1`] = `
<Text
  color="surface"
  style={
    Array [
      Object {
        "color": "#000",
        "fontFamily": "OpenSans-Regular",
        "fontSize": 14,
      },
    ]
  }
  theme={
    Object {
      "borders": Object {
        "radius": Object {
          "sm": 10,
          "xs": 4,
        },
      },
      "colors": Object {
        "background": Object {
          "main": "#fff",
          "onMain": "#000",
        },
        "error": Object {
          "main": "#D93936",
          "onMain": "#fff",
        },
        "lightGreen": Object {
          "main": "#95BFA7",
          "onMain": "#fff",
        },
        "orange": Object {
          "main": "#F28705",
          "onMain": "#fff",
        },
        "primary": Object {
          "main": "#3C9BA6",
          "onMain": "#fff",
        },
        "secondary": Object {
          "main": "#072B59",
          "onMain": "#fff",
        },
        "surface": Object {
          "main": "#000",
          "onMain": "#fff",
        },
        "surface100": Object {
          "main": "#4E4B59",
          "onMain": "#fff",
        },
        "surface50": Object {
          "main": "#E1E0E7",
          "onMain": "#fff",
        },
        "surface500": Object {
          "main": "#6B7785",
          "onMain": "#fff",
        },
        "surface600": Object {
          "main": "#41474D",
          "onMain": "#fff",
        },
      },
      "spacing": Object {
        "lg": 25,
        "md": 20,
        "sm": 10,
        "ty": 4,
        "xl": 38,
        "xs": 5,
        "xxl": 45,
        "xxxl": 100,
      },
      "typography": Object {
        "body1": Object {
          "fontFamily": "OpenSans-Regular",
          "fontSize": 14,
        },
        "body2": Object {
          "fontFamily": "OpenSans-Bold",
          "fontSize": 16,
        },
        "body3": Object {
          "fontFamily": "OpenSans-Regular",
          "fontSize": 14,
        },
        "caption": Object {
          "fontFamily": "OpenSans-Regular",
          "fontSize": 16,
        },
        "h1": Object {
          "fontFamily": "OpenSans-Light",
          "fontSize": 96,
        },
        "h2": Object {
          "fontFamily": "Montserrat-Bold",
          "fontSize": 48,
        },
        "h3": Object {
          "fontFamily": "Montserrat-Bold",
          "fontSize": 32,
        },
        "h4": Object {
          "fontFamily": "Montserrat-SemiBold",
          "fontSize": 32,
        },
        "h5": Object {
          "fontFamily": "Montserrat-Bold",
          "fontSize": 24,
        },
        "h6": Object {
          "fontFamily": "Montserrat-Regular",
          "fontSize": 24,
        },
        "h7": Object {
          "fontFamily": "Montserrat-SemiBold",
          "fontSize": 24,
        },
        "h8": Object {
          "fontFamily": "Montserrat-SemiBold",
          "fontSize": 20,
        },
        "overline": Object {
          "fontFamily": "OpenSans-SemiBold",
          "fontSize": 14,
        },
        "subtitle1": Object {
          "fontFamily": "OpenSans-Regular",
          "fontSize": 18,
        },
        "subtitle2": Object {
          "fontFamily": "Montserrat-Bold",
          "fontSize": 18,
        },
      },
    }
  }
  typography="body1"
>
  Hello world test
</Text>
`;
